### ISSUE TITLE
All types which are convertible to bool should be coercible to bool.

### ISSUE THREAD

**Issue by mlucy** [rethinkdb/rethinkdb#3986]
```rb
irb(main):199:0> r({}).coerce_to('bool').run
RethinkDB::RqlRuntimeError: Cannot coerce OBJECT to BOOL.
```

**Comment by danielmewes** [comment]
I think this is a duplicate of https://github.com/rethinkdb/rethinkdb/issues/3133

Is there a reason you put this in 2.0? Do you think it's very important?

**Comment by mlucy** [comment]
Ah, I missed #3133 .  This issue does seem to be identical.  I'd call this a bug, personally; I opened it because I was writing a query for someone on stackoverflow and had to write `.not().not()` which seemed opaque.  It's a pretty small fix, but leaving it in 2.1 seems fine to me.
